[BUG]: ValueError: x and y must have same first dimension, but have shapes (37,) and (38,)

### Describe the issue:

Running pytest on October 27th (daylight saving time change) creates an error.

### Reproduceable code example:

_No response_

### Error message:

```shell
Run pip install -e .
Obtaining file:///home/runner/work/EOS/EOS
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Checking if build backend supports build_editable: started
  Checking if build backend supports build_editable: finished with status 'done'
  Getting requirements to build editable: started
  Getting requirements to build editable: finished with status 'done'
  Preparing editable metadata (pyproject.toml): started
  Preparing editable metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: akkudoktor-eos
  Building editable for akkudoktor-eos (pyproject.toml): started
  Building editable for akkudoktor-eos (pyproject.toml): finished with status 'done'
  Created wheel for akkudoktor-eos: filename=akkudoktor_eos-0.0.1-0.editable-py3-none-any.whl size=7024 sha256=5a7b26abb4212100595c3e3a288ab517c716a42778fe56dc721e40ecb0fc4bce
  Stored in directory: /tmp/pip-ephem-wheel-cache-tr9l62_1/wheels/ac/b0/79/a49baa3d09bf3b62e73b8dda5e7952e8210221691da7a299a5
Successfully built akkudoktor-eos
Installing collected packages: akkudoktor-eos
Successfully installed akkudoktor-eos-0.0.1
============================= test session starts ==============================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0 -- /opt/hostedtoolcache/Python/3.12.7/x64/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/EOS/EOS
configfile: pyproject.toml
plugins: xprocess-1.0.2, cov-5.0.0
collecting ... collected 62 items

tests/test_cachefilestore.py::test_generate_cache_file_key PASSED
tests/test_cachefilestore.py::test_get_file_path PASSED
tests/test_cachefilestore.py::test_create_cache_file PASSED
tests/test_cachefilestore.py::test_get_cache_file PASSED
tests/test_cachefilestore.py::test_set_custom_file_object PASSED
tests/test_cachefilestore.py::test_delete_cache_file PASSED
2024-10-27 19:57:33,651 - /home/runner/work/EOS/EOS/src/akkudoktoreos/cachefilestore.py - WARNING - The cache file with key '3946a93846426ba0ce8fc02237bddbac3edccc9f4ad4a285122165be3ec1d401' is an in memory file object. Will only delete store entry but not file.
tests/test_cachefilestore.py::test_clear_all_cache_files PASSED
tests/test_cachefilestore.py::test_clear_cache_files_by_date PASSED
tests/test_cachefilestore.py::test_cache_file_with_date PASSED
tests/test_cachefilestore.py::test_recreate_existing_cache_file PASSED
tests/test_cachefilestore.py::test_cache_store_is_singleton PASSED
tests/test_cachefilestore.py::test_cache_in_file_decorator_caches_function_result PASSED
 * Debug mode: on

WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

 * Running on all addresses (0.0.0.0)

 * Running on http://127.0.0.1:8503

 * Running on http://10.1.0.17:8503

Press CTRL+C to quit

 * Restarting with stat

 * Debugger is active!

 * Debugger PIN: 187-337-542

akkudoktoreosserver process startup detected
PASSED
2024-10-27 19:57:53,666 - akkudoktoreos.class_pv_forecast - ERROR - Failed to load data from `http://127.0.0.1:8503/invalid?`. Status Code: 404
tests/test_pv_forecast.py::test_load_data_from_url /home/runner/work/EOS/EOS/.pytest_cache/d/.xprocess/akkudoktoreosserver$ /opt/hostedtoolcache/Python/3.12.7/x64/bin/python -m akkudoktoreosserver.flask_server
process 'akkudoktoreosserver' started pid=1842
 * Serving Flask app 'flask_server'

 * Debug mode: on

WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

 * Running on all addresses (0.0.0.0)

 * Running on http://127.0.0.1:8503

 * Running on http://10.1.0.17:8503

Press CTRL+C to quit

 * Restarting with stat

 * Debugger is active!

 * Debugger PIN: 187-337-542

akkudoktoreosserver process startup detected
PASSED
2024-10-27 19:58:01,662 - akkudoktoreos.class_pv_forecast - ERROR - Failed to load data from `http://127.0.0.1:8503/invalid?`. Status Code: 404
tests/test_pv_forecast.py::test_load_data_from_url_with_caching /home/runner/work/EOS/EOS/.pytest_cache/d/.xprocess/akkudoktoreosserver$ /opt/hostedtoolcache/Python/3.12.7/x64/bin/python -m akkudoktoreosserver.flask_server
process 'akkudoktoreosserver' started pid=1861
 * Serving Flask app 'flask_server'

 * Debug mode: on

WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

 * Running on all addresses (0.0.0.0)

 * Running on http://127.0.0.1:8503

 * Running on http://10.1.0.17:8503

Press CTRL+C to quit

 * Restarting with stat

 * Debugger is active!

 * Debugger PIN: 187-337-542

akkudoktoreosserver process startup detected
PASSED
tests/test_pv_forecast.py::test_report_ac_power_and_measurement PASSED
tests/test_pv_forecast.py::test_timezone_behaviour PASSED
tests/test_server.py::test_server /home/runner/work/EOS/EOS/.pytest_cache/d/.xprocess/akkudoktoreosserver$ /opt/hostedtoolcache/Python/3.12.7/x64/bin/python -m akkudoktoreosserver.flask_server
process 'akkudoktoreosserver' started pid=1885
 * Serving Flask app 'flask_server'

 * Debug mode: on

WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

 * Running on all addresses (0.0.0.0)

 * Running on http://127.0.0.1:8503

 * Running on http://10.1.0.17:8503

Press CTRL+C to quit

 * Restarting with stat

 * Debugger is active!

 * Debugger PIN: 187-337-542

akkudoktoreosserver process startup detected
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/coverage/inorout.py:505: CoverageWarning: Module modules was never imported. (module-not-imported)
  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")
PASSED

=================================== FAILURES ===================================
_________ test_optimize[optimize_input_1.json-optimize_result_1.json] __________

fn_in = 'optimize_input_1.json', fn_out = 'optimize_result_1.json'

    @pytest.mark.parametrize("fn_in, fn_out", [("optimize_input_1.json", "optimize_result_1.json")])
    def test_optimize(fn_in, fn_out):
        # Load input and output data
        with open(DIR_TESTDATA / fn_in, "r") as f_in:
            input_data = json.load(f_in)
    
        with open(DIR_TESTDATA / fn_out, "r") as f_out:
            expected_output_data = json.load(f_out)
    
        opt_class = optimization_problem(
            prediction_hours=48, strafe=10, optimization_hours=24, fixed_seed=42
        )
        start_hour = 10
    
        # Call the optimization function
>       ergebnis = opt_class.optimierung_ems(parameter=input_data, start_hour=start_hour, ngen=3)

tests/test_class_optimize.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/akkudoktoreos/class_optimize.py:458: in optimierung_ems
    visualisiere_ergebnisse(
src/akkudoktoreos/visualize.py:111: in visualisiere_ergebnisse
    plt.plot(
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/matplotlib/pyplot.py:3794: in plot
    return gca().plot(
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/matplotlib/axes/_axes.py:1779: in plot
    lines = [*self._get_lines(self, *args, data=data, **kwargs)]
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/matplotlib/axes/_base.py:296: in __call__
    yield from self._plot_args(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <matplotlib.axes._base._process_plot_var_args object at 0x7f8dfb0846b0>
axes = <Axes: >
tup = [array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35,....01, 1257.98,
       1232.67,  871.26,  860.88, 1158.03, 1222.72, 1221.04,  949.99,
        987.01,  733.99,  592.97])]
kwargs = {'alpha': 0.8, 'label': 'Load (Wh)', 'linestyle': '-', 'marker': 'o'}

    def _plot_args(self, axes, tup, kwargs, *,
                   return_kwargs=False, ambiguous_fmt_datakey=False):
        """
        Process the arguments of ``plot([x], y, [fmt], **kwargs)`` calls.
    
        This processes a single set of ([x], y, [fmt]) parameters; i.e. for
        ``plot(x, y, x2, y2)`` it will be called twice. Once for (x, y) and
        once for (x2, y2).
    
        x and y may be 2D and thus can still represent multiple datasets.
    
        For multiple datasets, if the keyword argument *label* is a list, this
        will unpack the list and assign the individual labels to the datasets.
    
        Parameters
        ----------
        tup : tuple
            A tuple of the positional parameters. This can be one of
    
            - (y,)
            - (x, y)
            - (y, fmt)
            - (x, y, fmt)
    
        kwargs : dict
            The keyword arguments passed to ``plot()``.
    
        return_kwargs : bool
            Whether to also return the effective keyword arguments after label
            unpacking as well.
    
        ambiguous_fmt_datakey : bool
            Whether the format string in *tup* could also have been a
            misspelled data key.
    
        Returns
        -------
        result
            If *return_kwargs* is false, a list of Artists representing the
            dataset(s).
            If *return_kwargs* is true, a list of (Artist, effective_kwargs)
            representing the dataset(s). See *return_kwargs*.
            The Artist is either `.Line2D` (if called from ``plot()``) or
            `.Polygon` otherwise.
        """
        if len(tup) > 1 and isinstance(tup[-1], str):
            # xy is tup with fmt stripped (could still be (y,) only)
            *xy, fmt = tup
            linestyle, marker, color = _process_plot_format(
                fmt, ambiguous_fmt_datakey=ambiguous_fmt_datakey)
        elif len(tup) == 3:
            raise ValueError('third arg must be a format string')
        else:
            xy = tup
            linestyle, marker, color = None, None, None
    
        # Don't allow any None value; these would be up-converted to one
        # element array of None which causes problems downstream.
        if any(v is None for v in tup):
            raise ValueError("x, y, and format string must not be None")
    
        kw = {}
        for prop_name, val in zip(('linestyle', 'marker', 'color'),
                                  (linestyle, marker, color)):
            if val is not None:
                # check for conflicts between fmt and kwargs
                if (fmt.lower() != 'none'
                        and prop_name in kwargs
                        and val != 'None'):
                    # Technically ``plot(x, y, 'o', ls='--')`` is a conflict
                    # because 'o' implicitly unsets the linestyle
                    # (linestyle='None').
                    # We'll gracefully not warn in this case because an
                    # explicit set via kwargs can be seen as intention to
                    # override an implicit unset.
                    # Note: We don't val.lower() != 'none' because val is not
                    # necessarily a string (can be a tuple for colors). This
                    # is safe, because *val* comes from _process_plot_format()
                    # which only returns 'None'.
                    _api.warn_external(
                        f"{prop_name} is redundantly defined by the "
                        f"'{prop_name}' keyword argument and the fmt string "
                        f'"{fmt}" (-> {prop_name}={val!r}). The keyword '
                        f"argument will take precedence.")
                kw[prop_name] = val
    
        if len(xy) == 2:
            x = _check_1d(xy[0])
            y = _check_1d(xy[1])
        else:
            x, y = index_of(xy[-1])
    
        if axes.xaxis is not None:
            axes.xaxis.update_units(x)
        if axes.yaxis is not None:
            axes.yaxis.update_units(y)
    
        if x.shape[0] != y.shape[0]:
>           raise ValueError(f"x and y must have same first dimension, but "
                             f"have shapes {x.shape} and {y.shape}")
E           ValueError: x and y must have same first dimension, but have shapes (37,) and (38,)

/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/matplotlib/axes/_base.py:486: ValueError

---------- coverage: platform linux, python 3.12.7-final-0 -----------
Name                                         Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------
src/akkudoktoreos/__init__.py                    0      0   100%
src/akkudoktoreos/cachefilestore.py            217     31    86%   247, 301, 303-304, 306, 314, 354, 358, 360, 393, 395, 397, 411-416, 423-424, 476-477, 483-484, 562-564, 581-582, 596, 599-600, 621-622
src/akkudoktoreos/class_akku.py                105     38    64%   35, 50-65, 108, 133, 163-228
src/akkudoktoreos/class_ems.py                  84      7    92%   45, 52-54, 101-105
src/akkudoktoreos/class_haushaltsgeraet.py      25      4    84%   19, 21, 48, 56
src/akkudoktoreos/class_inverter.py             34      7    79%   15-18, 33-34, 52
src/akkudoktoreos/class_load.py                 47     47     0%   1-102
src/akkudoktoreos/class_load_container.py       17     17     0%   1-39
src/akkudoktoreos/class_load_corrector.py       90     90     0%   1-146
src/akkudoktoreos/class_numpy_encoder.py        12     12     0%   1-25
src/akkudoktoreos/class_optimize.py            160     30    81%   71, 106, 122-133, 138-144, 157, 163, 200, 209, 214, 229, 264, 271, 275-278, 296-297, 327, 344, 452, 474-502
src/akkudoktoreos/class_pv_forecast.py         230     24    90%   173, 348-365, 369, 375, 394, 403, 427, 442, 655-671
src/akkudoktoreos/class_soc_calc.py            169    169     0%   1-340
src/akkudoktoreos/class_sommerzeit.py            8      0   100%
src/akkudoktoreos/class_strompreis.py           81     81     0%   1-128
src/akkudoktoreos/config.py                     13      6    54%   28-34
src/akkudoktoreos/datetimeutil.py               93     18    81%   70, 73, 102-109, 120, 124-127, 144, 197, 213, 220, 225, 265
src/akkudoktoreos/heatpump.py                   57     22    61%   80-82, 101-103, 107-117, 122-137
src/akkudoktoreos/logutil.py                    32      2    94%   60, 85
src/akkudoktoreos/visualize.py                 153    100    35%   106, 119-356
--------------------------------------------------------------------------
TOTAL                                         1627    705    57%

=========================== short test summary info ============================
FAILED tests/test_class_optimize.py::test_optimize[optimize_input_1.json-optimize_result_1.json] - ValueError: x and y must have same first dimension, but have shapes (37,) and (38,)
======================== 1 failed, 61 passed in 40.27s =========================
pytest-xprocess reminder::Be sure to terminate the started process by running 'pytest --xkill' if you have not explicitly done so in your fixture with 'xprocess.getinfo(<process_name>).terminate()'.
Error: Process completed with exit code 1.
```


### Version information:

main on cdf4d074e211cf689bf3b91063aba6d5bcd98047

--- a/src/akkudoktoreos/class_sommerzeit.py
+++ b/src/akkudoktoreos/class_sommerzeit.py
@@ -2,6 +2,7 @@ import datetime
 import zoneinfo
 
 
+# currently unused
 def ist_dst_wechsel(tag: datetime.datetime, timezone="Europe/Berlin") -> bool:
     """Checks if Daylight Saving Time (DST) starts or ends on a given day."""
     tz = zoneinfo.ZoneInfo(timezone)

--- a/src/akkudoktoreos/visualize.py
+++ b/src/akkudoktoreos/visualize.py
@@ -1,12 +1,9 @@
-import datetime
-
 # Set the backend for matplotlib to Agg
 import matplotlib
 import matplotlib.pyplot as plt
 import numpy as np
 from matplotlib.backends.backend_pdf import PdfPages
 
-from akkudoktoreos.class_sommerzeit import ist_dst_wechsel
 from akkudoktoreos.config import AppConfig, SetupIncomplete
 
 matplotlib.use("Agg")
@@ -100,11 +97,7 @@ def visualisiere_ergebnisse(
         #####################
 
         plt.figure(figsize=(14, 10))
-
-        if ist_dst_wechsel(datetime.datetime.now()):
-            hours = np.arange(start_hour, config.eos.prediction_hours - 1)
-        else:
-            hours = np.arange(start_hour, config.eos.prediction_hours)
+        hours = np.arange(start_hour, config.eos.prediction_hours)
 
         # Energy flow, grid feed-in, and grid consumption
         plt.subplot(3, 2, 1)
@@ -216,10 +209,7 @@ def visualisiere_ergebnisse(
         ax1.set_title("AC/DC Charging and Discharge Overview")
         ax1.grid(True)
 
-        if ist_dst_wechsel(datetime.datetime.now()):
-            hours = np.arange(start_hour, config.eos.prediction_hours - 1)
-        else:
-            hours = np.arange(start_hour, config.eos.prediction_hours)
+        hours = np.arange(start_hour, config.eos.prediction_hours)
 
         pdf.savefig()  # Save the current figure state to the PDF
         plt.close()  # Close the current figure to free up memory

